semaphore/integration_tests.py

```diff
@@ -87,37 +89,63 @@ def get_trip_plans(environment, from_place, to_place, trip_date, arrive_by):
     return requests.get(url, params).json()
 
 
-# alerts come in random order, which sometimes causes plan comparison to fail
-def sort_alerts(json):
-    plan = json.get("plan")
-    for itinerary in plan.get("itineraries"):
-        for leg in itinerary.get("legs"):
-            alerts = leg.get("alerts")
-            if alerts:
-                leg["alerts"] = sorted(alerts, key=lambda a: a["alertDescriptionText"])
+def process_plan(plan):
+    # alerts come in random order, which sometimes causes plan comparison to fail
+    def sort_alerts(json):
+        plan = json.get("plan")
+        for itinerary in plan.get("itineraries"):
+            for leg in itinerary.get("legs"):
+                alerts = leg.get("alerts")
+                if alerts:
+                    leg["alerts"] = sorted(alerts, key=lambda a: a["alertDescriptionText"])
+        return json
+
+    # temporary stops have random IDs
+    def remove_random_stop_ids(json):
+        if isinstance(json, list):
+            return [remove_random_stop_ids(j) for j in json]
+        elif isinstance(json, dict):
+            result = {}
+            for key, value in json.items():
+                if key == "stopId":
+                    result[key] = re.sub(r"(\w+:temp_0)\.\d+", r"\1-random", value)
+                else:
+                    result[key] = value
+            return result
```

Comment:
I think you can simplify this:
```python
if isinstance(json, list):
  return [remove_random_stop_ids(j) for j in json]

if isinstance(json, dict) and "stopId" in json:
  json = json.copy()
  json["stopId"] = re.sub(...)

return json
```

Reply:
I don't think it will process nested dictionaries correctly like that.

Reply:
You mean where `value` is a dictionary? I don't see that happening here, ether.

Reply:
Yep, I've fixed it in the latest commit.